File: web-app/src/lib/features/features/workspace.ts
import type { Feature } from '../types';

export const workspaceFeatures = {
  'workspace-get-info': {
    id: 'workspace-get-info',
    title: 'Get Workspace Info',
    description: 'Retrieves workspace metadata and configuration for a given session.',
    rpcIds: ['workspace:get-info'],
    componentPaths: [],
    testIds: [
      'TestGetWorkspaceInfo_EmptyID',
      'TestGetWorkspaceInfo_SessionNotFound',
      'TestGetWorkspaceInfo_NoSession',
    ],
    status: 'stable',
    since: '1.0.0',
  },
  'workspace-list-targets': {
    id: 'workspace-list-targets',
    title: 'List Workspace Targets',
    description: 'Lists available workspace targets that a session can switch to.',
    rpcIds: ['workspace:list-targets'],
    componentPaths: [],
    testIds: [
      'TestListWorkspaceTargets_EmptyID',
      'TestListWorkspaceTargets_SessionNotFound',
      'TestListWorkspaceTargets_NoConfig',
    ],
    status: 'stable',
    since: '1.0.0',
  },
  'workspace-switch': {
    id: 'workspace-switch',
    title: 'Switch Workspace',
    description: 'Switches a session to a different workspace target.',
    rpcIds: ['workspace:switch'],
    componentPaths: [],
    testIds: [
      'TestWorkspaceService_SwitchWorkspace_MissingID',
      'TestWorkspaceService_SwitchWorkspace_MissingTarget',
      'TestWorkspaceService_ConcurrentSwitchReturnsUnavailable',
      'TestWorkspaceService_SwitchGuardCleansUpOnCompletion',
      'TestWorkspaceService_SwitchGuardIsPerSession',
    ],
    status: 'stable',
    since: '1.0.0',
  },
} as const satisfies Record<string, Feature>;

File: server/features/workspace.go
package features

import "github.com/tstapler/stapler-squad/server/featureregistry"

// WorkspaceGetInfo describes the get-workspace-info RPC.
var WorkspaceGetInfo = featureregistry.Feature{
	ID:          "workspace-get-info",
	Title:       "Get Workspace Info",
	Description: "Retrieves workspace metadata and configuration for a given session.",
	RPCIDs:      []string{"workspace:get-info"},
	Status:      featureregistry.StatusStable,
	Since:       "1.0.0",
}

// WorkspaceListTargets describes the list-workspace-targets RPC.
var WorkspaceListTargets = featureregistry.Feature{
	ID:          "workspace-list-targets",
	Title:       "List Workspace Targets",
	Description: "Lists available workspace targets that a session can switch to.",
	RPCIDs:      []string{"workspace:list-targets"},
	Status:      featureregistry.StatusStable,
	Since:       "1.0.0",
}

// WorkspaceSwitch describes the switch-workspace RPC.
var WorkspaceSwitch = featureregistry.Feature{
	ID:          "workspace-switch",
	Title:       "Switch Workspace",
	Description: "Switches a session to a different workspace target.",
	RPCIDs:      []string{"workspace:switch"},
	Status:      featureregistry.StatusStable,
	Since:       "1.0.0",
}

func init() {
	featureregistry.Register(WorkspaceGetInfo)
	featureregistry.Register(WorkspaceListTargets)
	featureregistry.Register(WorkspaceSwitch)
}
